src/decorators.ts

```diff
@@ -194,7 +213,13 @@ function _query(
  * @param target A single element by id or EventTarget to target
  */
 export const listen = (eventName: string, target: string|EventTarget) =>
-    (proto: any, methodName: string) => {
+    (proto: ElementPrototype, methodName: string) => {
+      if (!proto.constructor._addDeclarativeEventListener) {
+        console.error(
+            `Cannot add listener for ${eventName} because ` +
+            `DeclarativeEventListeners mixin was not applied to element.`);
+        return;
+      }
       proto.constructor._addDeclarativeEventListener(
-          target, eventName, proto[methodName]);
+          target, eventName, (proto as any)[methodName]);
```

Comment:
I think we can remove this `any` too:

```typescript
<E extends ElementPrototype, K extends keyof E>(proto: E, methodName: K) => { /* ... */};
```

This doesn't enforce that `methodName` is a method (though we will be able to with 2.8!), but it does enforce that it's a property on `proto`

Reply:
I think we could do this a number of places in this file actually, e.g. with `observe`

Reply:
Discussed offline. Added note that the cast is safe here because it represents, by definition, the method we are decorating.